File: insertion_sort/insertion_sort.js
const insertion_sort = (arr) => {
  const size = arr.length - 1;
  for (let i = 0; i < size; i++) {
    let anchor = arr[i];
    let j = i - 1;
    while (j >= 0 && anchor < arr[j]) {
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = anchor;
  }
  return arr;
};

const elements = [5, 9, 2, 1, 67, 34, 88, 34];
console.log("Given Array : ");
console.log(elements);
console.log("Sorted Array : ");
console.log(insertion_sort(elements));

File: insertion_sort/insertion_sort.py
def insertion_sort(arr):
    size = len(arr) - 1

    for i in range(1, size):
        anchor = arr[i]
        j = i - 1 #always will be lesser then anchor's position
        while j >= 0 and anchor < arr[j]:
            arr[j+1] = arr[j] #shift all the greater values to the right
            j = j - 1
        arr[j+1] = anchor #replace with anchor value with suitable position
    
        

if __name__ == '__main__':
    elements = [5,9,2,1,67,34,88,34]
    print("Given array : " )
    print(elements)
    insertion_sort(elements)
    print("Sorted array : " )
    print(elements)